pfam_experiment.py

```diff
@@ -87,142 +127,171 @@ def main(_):
-                                   batch_size=FLAGS.batch_size)
-
-	lens_accuracy = results['accuracy']
-	lens_cross_entropy = float(results['cross_entropy'])
-
-	embedding_optimizer = create_optimizer(embedding_model, 
-										   learning_rate=FLAGS.learning_rate, 
-										   weight_decay=FLAGS.weight_decay, 
-										   layers=layers)
-
-	assert(embedding_optimizer.target.params.keys()==trained_params.keys()), 'Optimizer parameters do not match!'
-	for layer in embedding_optimizer.target.params.keys():
-		embedding_optimizer.target.params[layer] = trained_params[layer]
-
-	train_knn_results = pfam_nearest_neighbors_classification(encoder=embedding_optimizer.target, 
-                                                              train_family_accessions=lens_knn_train_family_accessions, 
-                                                              test_family_accessions=lens_knn_train_family_accessions,
-                                                              batch_size=FLAGS.batch_size,
-                                                              train_samples=FLAGS.knn_train_samples)[0]
-	train_knn_accuracy = train_knn_results['1-nn accuracy']
-
-	test_knn_results = pfam_nearest_neighbors_classification(encoder=embedding_optimizer.target, 
-                                                             train_family_accessions=knn_test_family_accessions, 
-                                                             test_family_accessions=knn_test_family_accessions,
-                                                             batch_size=FLAGS.batch_size,
-                                                             train_samples=FLAGS.knn_train_samples)[0]
-	test_knn_accuracy = test_knn_results['1-nn accuracy']
-
-	datum = {
-				'encoder_fn_name': FLAGS.encoder_fn_name,
-				'encoder_fn_kwargs_path': FLAGS.encoder_fn_kwargs_path,
-				'reduce_fn_name': FLAGS.reduce_fn_name,
-				'reduce_fn_kwargs_path': FLAGS.reduce_fn_kwargs_path,
-				'epochs': FLAGS.epochs,
-				'batch_size': FLAGS.batch_size,
-				'encoder_lr': FLAGS.learning_rate[0],
-				'lens_lr': FLAGS.learning_rate[1],
-				'predictor_lr': FLAGS.learning_rate[2],
-				'encoder_wd': FLAGS.weight_decay[0],
-				'lens_wd': FLAGS.weight_decay[1],
-				'predictor_wd': FLAGS.weight_decay[2],
-				'train_families': FLAGS.train_families,
-				'use_transformer': FLAGS.use_transformer,
-				'use_bert': FLAGS.use_bert,
-				'restore_transformer_dir': FLAGS.restore_transformer_dir,
-				'knn_train_samples': FLAGS.knn_train_samples,
-				'lens_cross_entropy': lens_cross_entropy,
-				'lens_accuracy': lens_accuracy,
-				'train_knn_accuracy': train_knn_accuracy,
-				'test_knn_accuracy': test_knn_accuracy
-			}
-	
 	print(datum)
 	df = pd.DataFrame([datum])
     
-	with gcsfs.open(os.path.join('sweep_data', 'experiment' + '.csv'), 'w') as gcs_file:
+	with gcsfs.open(os.path.join(FLAGS.save_dir, FLAGS.index + '.csv'), 'w') as gcs_file:
 		df.to_csv(gcs_file)
```

Comment:
may be helpful downstream to do to_csv(..., index=False)

Reply:
great catch, thanks